### PR TITLE
feat: turn into nexus plugin

### PR DESCRIPTION
**Pull request by jasonkuhrt**
#### TODO 
- [x] https://github.com/prisma-labs/nexus/pull/236

BREAKING CHANGE

- `NexusPrismaParams` is now exported as `Options`
- `nexusPrismaPlugin` is now a Nexus Plugin.

  * It was previously a NexusPrisma Builder class factory.
  * It no longer returns type definitions to be passed to `Nexus.makeSchema({ types })`.
  * Instead the return value should be passed into `Nexus.makeSchema({ plugins })`.

### CODE REVIEW

File: examples/blog/src/schema/index.ts [hunk 1]
```diff
@@ -8,11 +8,10 @@ import * as Post from './Post'
 import * as Author from './Author'
 
 const appTypes = [Query, Mutation, Blog, Post, Author]
-const nexusPrismaTypes = NexusPrisma.nexusPrismaPlugin({ types: appTypes })
-const allTypes = [appTypes, nexusPrismaTypes]
 
 export default Nexus.makeSchema({
-  types: allTypes,
+  types: appTypes,
+  plugins: [NexusPrisma.create()],
```
Comment (jasonkuhrt) [inline examples/blog/src/schema/index.ts]: Or `nexusPrismaPlugin()`.

I expect we're going to have a discussion about if we should enable two ways of doing it, and if only one way, which way.

File: src/index.ts [hunk 2]
```diff
@@ -1,4 +1,14 @@
+// import { Plugin } from 'nexus'
```
Comment (jasonkuhrt) [inline src/index.ts]: We will restore this before merging.

File: src/index.ts [hunk 3]
```diff
@@ -1,4 +1,19 @@
...
 import * as Builder from './builder'
 
+// TODO this should be called NexusPrismaOptions
```
Comment (jasonkuhrt) [inline src/index.ts]: Future PR

File: src/builder.ts [hunk 4]
```diff
@@ -41,6 +40,7 @@ export interface Options {
 
 export interface InternalOptions extends Options {
   dmmf?: DMMF.DMMF // For testing
+  types: any // FIXME `any`
```
Comment (jasonkuhrt) [inline src/builder.ts]: Another PR

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "nexus-prisma",
-  "version": "0.5.0-next.1",
+  "version": "0.0.0-pr.434.b880059",
```
Comment (jasonkuhrt) [inline package.json]: undo before merge

File: package.json [hunk 6]
```diff
@@ -19,7 +19,7 @@
   },
   "dependencies": {
     "fs-extra": "^8.1.0",
-    "nexus": "^0.12.0-beta.9",
+    "nexus": "pr236",
```
Comment (jasonkuhrt) [inline package.json]: undo before merge

File: examples/blog/package.json [hunk 7]
```diff
@@ -15,8 +15,8 @@
   "dependencies": {
     "graphql": "^14.3.1",
     "graphql-yoga": "^1.18.2",
-    "nexus": "^0.12.0-beta.8",
-    "nexus-prisma": "^0.4.1"
+    "nexus": "pr236",
+    "nexus-prisma": "^0.0.0-pr.434.b880059"
```
Comment (jasonkuhrt) [inline examples/blog/package.json:21]: change these before merging

File: src/builder.ts [hunk 8]
```diff
@@ -142,7 +109,7 @@ export class SchemaBuilder {
   /**
    * Build `t.crud` dynamic output property
    */
-  protected buildCRUD(): DynamicOutputPropertyDef<'crud'> {
+  buildCRUD(): DynamicOutputPropertyDef<'crud'> {
```
Comment (Weakky) [inline src/builder.ts:186]: What's the reasoning for making all class methods protected? (Both in the `SchemaBuilder` and the `Publisher`)

**Comment by Weakky** [comment]
Apart from my comment above, the rest looks great! 👍

File: src/builder.ts (hunk 8 above)
Comment (jasonkuhrt) [reply src/builder.ts:186]: > What's the reasoning for making all class methods protected?

Do you mean unprotected/public?

Yeah that change is unrelated to this PR, will revert here.

Comment (Weakky) [reply src/builder.ts:186]: Oh yeah sorry, I meant unprotected. Cool 👍

File: examples/blog/src/schema/index.ts (hunk 1 above)
Comment (jasonkuhrt) [reply examples/blog/src/schema/index.ts]: aborted this

**Pull request merged**